tests/behat/features/bootstrap/LibContext.php

```diff
@@ -0,0 +1,1399 @@
+    $fileNamePostfix = "";
+    while (file_exists($fileName . $fileNamePostfix . '.png')) {
+      $fileNamePostfix++;
+    }
+    $fileName = $fileName . $fileNamePostfix . '.png';
+
+    // Log the filename of the screenshot to notify the user.
+    print_r("Screenshot in: " . $fileName . "\n");
+
+    // Now find the actual height of the shown page.
+    $height = $this->minkContext->getSession()->evaluateScript("return document.body.scrollHeight;");
+    // Save the screenshot.
+    $this->minkContext->getSession()
+          ->getDriver()
+          ->resizeWindow(1280, $height, 'current');
+    file_put_contents($fileName, $this->minkContext->getSession()->getDriver()->getScreenshot());
+  }
+
+  /**
+   * Step
+   *
+   * @Given I accept cookies
+   *
+   * @throws Exception
+   *   Checks if cookie acceptance is shown, and accepts it if it is.
+   */
+  public function acceptCookiesMinimizeAskLibrarianOverlay() {
+    // We use the searchPage-instance to deal with cookies.
+    $this->check($this->searchPage->acceptCookiesMinimizeAskLibrarianOverlay(), $this->searchPage->getMessages());
+  }
+
+  /**
+   * General function to check the outcome of a function and handle the result
+   *
+   * @param string $result
+   *   If non-empty an exception is thrown.
+   * @param string $msg
+   *   The message to show together with the exception.
+   *
+   * @throws Exception
+   *   In case of error.
+   */
+  public function check($result, $msg = '') {
+    // Log messages if we have any.
+    if ($msg !== '') {
+      print_r($msg);
+    }
+    // Fail if we have a non-empty string.
+    if ($result !== "") {
+      throw new Exception($result);
+    }
+  }
+
+
+  /**
+   * Step
+   *
+   * @Then I check if the right number of search results are shown
+   *
+   * The methods "I use facets to reduce..." sets $this->>expectedResultsCount and must
+   * be used before this is called, otherwise it will fail with that message.
+   */
+  public function checkIfTheRightNumberOfSearchResultsAreShown() {
+    // Scrape off the search result size from the page. This is displayed on every search result page.
+    $resultSize = $this->searchPage->getShownSizeOfSearchResult();
+    if ($resultSize < 0) {
+      print_r($this->searchPage->getMessages());
+      throw new Exception("Couldn't find a search result size on page.");
+    }
+
+    // Now compare to the expected number.
+    if ($this->searchPage->getExpectedSearchResultSize() != 0) {
+      if ($this->searchPage->getExpectedSearchResultSize() != $resultSize) {
+        throw new Exception("Fandt ikke det forventede antal poster. (Fandt: " . $resultSize . ". Forventede:" . $this->searchPage->getExpectedSearchResultSize() . ")");
+      }
+    }
+    else {
+      throw new Exception("An expected number was never set. Use facets to set it.");
+    }
+  }
+
+  /**
+   * This is meant only to be used after a multipage search when pageing is in place.
```

Comment:
Please reformat to have proper DocBlock summary/description

Reply:
Done.